src/index/Index.cpp

```diff
@@ -444,6 +292,16 @@ void Index::createPermutation(const string& fileName, Index::ExtVec const& vec,
   out.close();
   LOG(INFO) << "Permutation done.\n";
 }
+template <class MetaData, class Comparator>
+void Index::createPermutationWrapper(
```

Comment:
So createPermutationWrapper -> createPermutation
createPermutation -> createPermutationImpl

I would keep it in the header since in my understanding the convention is, that the public functions come first and also form some type of interface documentation whereas the private functions are there for the previous definitions. Additionally the createPermutation() (low-level) has to be declared in the header since it is not static anymore.

Reply:
Sounds good. I thought it doesn't have to be in the header since it's only used in that one `.cpp` file?

Reply:
It is a class member of Index. And I thought that class members have to appear in the Declaration which is in the header.

Reply:
I also have wrapped the somewhat lengthy call to createPatterns in a similar way.